platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md

```diff
@@ -0,0 +1,203 @@
+state when restarted. However, this does not solve the problem if a deterministic crash affects all nodes, and it does
+not get us closer to the end-state of a simpler system.
+
+---
+
+## Changes
+
+### Architecture and/or Components
+
+At a high level, the architecture remains the same: events are created by the "Event Creator System", they are gossiped
+by the "Gossip System", events are handled by the "Event Intake System", persisted by the
+"Pre-Consensus Recording System", and come to consensus in the "Consensus System".
+
+![consensusPlatformArch](consensusPlatformArch.png)
+
+In this case, we are only interested in _self events_. These are created by the `Event Creator System` and passed to
+the `Event Intake System`. Here, they are ordered and validated and sent to the `Preconsensus Recording System`. Here,
+they will be immediately persisted using `MemoryMappedFile` and then sent to the `Consensus System` and to the
+`Gossip System`.
+
+In the short term, we will not send any events to the gossip system before they pass through the PCES system. When we
+eventually deploy gossip algorithms that call for out of order transmission (e.g. chatter), we may decide to expose
+other events (i.e. events not created by ourselves) to gossip before they pass through the orphan buffer.
+
+### Impact on Wiring
+
+#### PCES Writer
+
+PCES Writer will be changed to use `MemoryMappedFile`. The `flush` API will be removed (or, at least, no longer used
+by the consensus engine. It may be required for clean shutdown, which will be handled by another issue).
+
+#### Asynchronous writing changes
+
+Currently, PCES writing happens asynchronously. That is, an event is inserted into the PCES writer, and while
+the event is in the process of being written, other parts of the system utilize it. When it comes time to handle
+transactions in a round, we pause if the keystone event for that round has not yet been made durable on disk.
+
+After this change is made, no part of the system after event intake will utilize an event until it has been made
+durable by the PcesWriter, except for Gossip. Gossip will only have to wait for self-events to be persisted. 
+
+#### PcesSequencer removal
+
+The `PcesSequencer` will no longer be necessary and can be removed. There will also be no need for keystone events
+within rounds.
+
+#### RoundDurabilityBuffer removal
+
+The `RoundDurabilityBuffer` will no longer be necessary and can be removed.
+
+#### Diagram Changes
+
+In this image, several existing components are crossed out. This represents a component being removed. The data will
+still flow along the same pathways, but with no component on that wire.
+
+![](inlinePces.png)](inlinePces.png)
+
+### Core Behaviors
+
+[JEP 352](https://openjdk.org/jeps/352) enhanced the support for `MemoryMappedFile` to support the semantics we need
+for non-volatile memory:
```

Comment:
Jasper has been doing more digging, and I'm not sure whether this `ExtendedMapMode.READ_WRITE_SYNC` is right. Or even whether memory mapped files is necessary, or whether a basic FileOutputStream flushing on every self-event is sufficient. I wonder whether the architecture doc should leave off the implementation detail of the PCES box itself, but just say "this box will persist before gossiping self events" or "this box will persist all events before sending them to the hashgraph" and then leave room in the implementation for whatever is found to work.